Tests/Reqnroll.VisualStudio.Tests/Discovery/DiscoveryTests.cs

```diff
@@ -84,9 +78,7 @@ public void DoNotTriggersCacheUpdateOnEventsForTheSameProject(LabeledTestData<Ac
         triggerEvent.Data(sut);
 
         //assert
-        sut.BindingRegistryCache.Verify(c =>
-                c.Update(It.IsAny<Func<ProjectBindingRegistry, Task<ProjectBindingRegistry>>>()), Times.Once,
-            "the cache update have to be called only once when the project haven't changed");
+        sut.BindingRegistryCache.Received(1).Update(Arg.Any<Func<ProjectBindingRegistry, Task<ProjectBindingRegistry>>>()); // the cache update have to be called only once when the project haven't changed
```

Comment:
Would it be worth the effort to ensure that the error message here gets output when the Received constraint fails?
Something perhaps along the lines of: 
```c#
Action assertion = () => sut.BindingRegistryCache.Received(1).Update(Arg.Any<Func<ProjectBindingRegistry, Task<ProjectBindingRegistry>>>());
assertion.Should().NotThrow("the cache update have to be called only once when the project haven't changed");
```